Add a test to ensure styles aren't duplicated

DanielTurner opened the pull request:
We have a polymer 3 app that has been having some issues.

In certain circumstances on an unusual setup (it includes dynamic DOM injection) you might have a situation that leads to the following: 
Failed to execute 'define' on 'CustomElementRegistry': this name has already been used with this registry

Only vaadin material elements have this issue: Polymer elements seem to be immune.
This test will ensure that duplicates don't happen.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-material-styles/88)
<!-- Reviewable:end -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/vaadin-material-styles?pullRequest=88) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/vaadin-material-styles?pullRequest=88) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vaadin/vaadin-material-styles?pullRequest=88) it.</sub>

web-padawan commented:
It looks like you are loading 2 different versions of the Material styled components, which might result in some other components, except this one, to be also registered twice.

Consider doing the following things first:

1. Check if any component brings its own `vaadin-material-styles` version:

```sh
npm ls | grep vaadin-material-styles
```

The output should look e.g. as follows:

```console
$ npm ls | grep vaadin-material-styles
│ │ │ ├── @vaadin/vaadin-material-styles@1.2.0 deduped
│ │ ├─┬ @vaadin/vaadin-material-styles@1.2.0
│ │ │ ├── @vaadin/vaadin-material-styles@1.2.0 deduped
```

2. If there are different versions without "deduped" mark, run the command:

```sh
npm dedupe
```

If that doesn't help, try to check if the bundler you are using is configured properly.

See also Polymer/polymer#5407 for similar error.

DanielTurner commented:
Thanks for your help, I have already done this, every time I run npm dedupe it ends up returning 404's so I guess with what you are saying its the bundler.

web-padawan commented:
@DanielTurner could you provide more information on your setup so that we could investigate?

The suggested workaround only removes the symptom, and I think we should fix the root issue instead.

DanielTurner commented:
How we fixed the issue:
Firstly we had different versions, but this was due to our package.json having something like this:
```    "@vaadin/vaadin-material-styles": "@vaadin/vaadin-material-styles",```
Which for some reason pointed to version 1.2.0 instead of 1.2.2, we have changed it to ^1.2.2.

The reason our bundler (polymer) didn't pick it up is probably because we have dynamic injection earlier on in the app and it doesn't understand what is going on. So we have to fill polymer.json with extraDependencies and in this case even run build before it works.

The solution is either don't do dynamic injection or get a better bundler. I really think ours is an edge case not many will have.

web-padawan commented:
> The solution is either don't do dynamic injection or get a better bundler. I really think ours is an edge case not many will have.

Agree that this should be resolved by making sure you only load each specific `@vaadin` dependency once, which should be doable with most of bundlers.

Closing per above comment and because there is nothing to fix on our side here.
